arlo_server/routes.py

```diff
@@ -1333,10 +1343,18 @@ def pretty_affiliation(affiliation):
 @app.route("/election/<election_id>/audit/reset", methods=["POST"])
 def audit_reset(election_id):
     # deleting the election cascades to all the data structures
-    Election.query.filter_by(id=election_id).delete()
-    db.session.commit()
+    election = Election.query.filter_by(id=election_id).one_or_none()
+    if election:
+        db.session.delete(election)
+        db.session.commit()
 
-    create_election(election_id)
+    # TODO election.audit_name doesn't exist if we didn't find an election for election_id
```

Comment:
I could use help understanding what to do in this case @benadida

Reply:
My proposal would be to just return a 404 right off the bat. The behavior of creating a new election with the ID of the old one even if there never was an old one just seemed like a byproduct of a SQL `DELETE` call operating on zero records not being an error, rather than an intentional choice.

Reply:
Nevermind, I don't know where I got the idea to do `one_or_none`... I thought the old code also checked for this but I must have misread

Reply:
It had the same effect since the `delete` would succeed even if it deleted 0 rows.

Reply:
added 404 here: 3c144d9